### PR TITLE
Implement SIGTERM graceful shutdown if pid is 1

### PR DESCRIPTION
**Pull request by jkroepke**
Related: https://blog.ghaiklor.com/avoid-running-nodejs-as-pid-1-under-docker-images-when-running-them-on-mesosphere-kubernetes-or-b7bd505657f9

Like in the blog described node has some signal issues if it runs as pid 1 (usually inside a container).
Currently on docker stop docker sends SIGTERM to node which will be ignored because not handled somewhere.

Haraka on docker or even on kubernetes has to be killed unless you use workarounds like `tini`. I would prefer that haraka initiate a graceful shutdown on docker stop.

Changes proposed in this pull request:
- Implement SIGTERM graceful shutdown if pid is 1

Checklist:
- [ ] docs updated
- [ ] tests updated
- [x] [Changes](https://github.com/haraka/Haraka/blob/master/Changes.md) updated

### CODE REVIEW

**Comment by jkroepke** [comment]
Did I something wrong? -- tests are broken.

**Comment by msimerson** [comment]
If you follow the links to the test output, you'll find the issue:

````
/home/travis/build/haraka/Haraka/haraka.js
  41:5  error  'signals' is never reassigned. Use 'const' instead  prefer-const
  43:1  error  Expected space(s) after "if"                        keyword-spacing
````

File: haraka.js [hunk 1]
```diff
@@ -38,7 +38,13 @@ process.on('uncaughtException', err => {
 });
 
 let shutting_down = false;
-['SIGINT'].forEach((sig) => {
+let signals = ['SIGINT'];
```
Comment (msimerson) [inline haraka.js]:
```suggestion
const signals = ['SIGINT'];
```

File: haraka.js [hunk 2]
```diff
@@ -38,7 +38,13 @@ process.on('uncaughtException', err => {
...
+
+if(process.pid === 1) {
```
Comment (msimerson) [inline haraka.js]:
```suggestion
if (process.pid === 1) {
```

**Comment by jkroepke** [comment]
I force-pushed into the branch with lint fixes applied. I also add a notice at the Changes.md

**Pull request merged**